blockchain/fullblocktests/generate.go

```diff
@@ -1089,6 +1090,80 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrFreshStakeMismatch)
 
+	// Attempt to add block with tickets voting on the wrong block
+	//
+	//   ... -> b36(8)
+	//                \-> bv6(9)
+	g.SetTip("b36")
+	g.NextBlock("bv6", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		wrongBlockScript, _ := hex.DecodeString("6a24008e029f92ae880d45ae61a5366b" +
+			"b81d9903c5e61045c5b17f1bc97260f8e54497000000")
+		b.STransactions[0].TxOut[0].PkScript = wrongBlockScript
+	})
+	rejected(blockchain.ErrVotesOnWrongBlock)
+
+	// Attempt to add block with incorrect votebits set
+	// Everyone votes Yea, but block header says Nay
+	//
+	//   ... -> b36(8)
+	//                \-> bv7(9)
+	g.SetTip("b36")
+	g.NextBlock("bv7", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		b.Header.VoteBits &= 0xFFFE
+	})
+	rejected(blockchain.ErrIncongruentVotebit)
+
+	// Attempt to add block with incorrect votebits set
+	// Everyone votes Nay, but block header says Yea
+	//
+	//   ... -> b36(8)
+	//                \-> bv8(9)
+	g.SetTip("b36")
+	g.NextBlock("bv8", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		b.Header.VoteBits = 0x0001
+		for i, stx := range b.STransactions {
```

Comment:
Rather than repeating the pattern over and over, please create a function similar to `ReplaceVotes` from `chaingen` which accepts a specific vote number, version, and bits.  I would suggest the following:

```Go
// ReplaceVoteBitsN returns a function that itself takes a block and modifies
// it by replacing the vote bits of the vote located at the provided index.  It
// will panic if the stake transaction at the provided index is not already a
// vote.
//
// NOTE: This must only be used as a munger to the 'NextBlock' function or it
// will lead to an invalid live ticket pool.
func ReplaceVoteBitsN(voteNum int, voteBits uint16) func(*wire.MsgBlock) {
	return func(b *wire.MsgBlock) {
		// Attempt to prevent misuse of this function by ensuring the
		// provided stake transaction number is actually a vote.
		stx := b.STransactions[voteNum]
		if !isVoteTx(stx) {
			panic(fmt.Sprintf("attempt to replace non-vote "+
				"transaction #%d for for block %s", voteNum,
				b.BlockHash()))
		}

		// Extract the existing vote version.
		existingScript := stx.TxOut[1].PkScript
		var voteVersion uint32
		if len(existingScript) >= 8 {
			voteVersion = binary.LittleEndian.Uint32(existingScript[4:8])
		}

		stx.TxOut[1].PkScript = voteBitsScript(voteBits, voteVersion)
	}
}
```

That way it uses the existing infrastructure to properly recreate the entire vote script and will panic if it's used improperly to help ensure the tests stay valid.

Then this function will read more like:
```Go
b.Header.VoteBits |= 0x0001

const voteBitsNo = 0x000
for i := 0; i < 4; i++ {
	chaingen.ReplaceVoteBitsN(i, voteBitsNo)
}
```